### DISCUSSION TITLE
workspace scope limitations

### DISCUSSION THREAD

**Discussion by tangweisuper** [eclipse-theia/theia#14122] [Q&A] [answered]
### Feature Description:
I'm building a web ide for developers(It is possible that they are untrusted developers). I hope developers can only see their code（/home/project）.But in fact, theia can go anywhere.

i want this
![image](https://github.com/user-attachments/assets/c7b46121-8a95-4265-bce4-c64dcb93a544)

but i can go there
![image](https://github.com/user-attachments/assets/f7f34377-891b-4eda-b605-0aa3fcef9539)

even /etc
![image](https://github.com/user-attachments/assets/99d16ea1-c48a-4de7-92af-a46d529c0823)

and this result can also be achieved by using the "Add Floder To Workspace" function
![image](https://github.com/user-attachments/assets/16533fd0-3d44-4381-9179-f4010d76bc35)

I already use filters to disable terminal functions, Is there a way to solve the above problem ?
thx

**Comment by msujew** [comment, marked as answer]
Hey @tangweisuper,

I've converted this into a discussion, since it's mostly a question and not a feature request.

Essentially, similarly to all other IDEs, Theia is built in a way that allows it to read/open all files that are accessible to the system user that starts the Theia backend process. Changing this behavior is pretty difficult, and even if you change the frontend/file system implementation behavior enough, you still have the issue of users having unrestricted access via the terminal and/or VS Code extensions.

The most common way to solve this is by isolating the IDE from the rest of the system. Most commonly, you would set something up like [Theia Cloud](https://theia-cloud.io/) or a similar, pod/container based solution. Even if users make their way out of the designated project area, they have no possibility of breaking something for themselves or in the infrastructure as they cannot escape the container.
